File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/web/models/configurations/presets/PresetProperty.java
package com.palitri.openiot.construction.framework.web.models.configurations.presets;

import com.palitri.openiot.construction.framework.web.models.configurations.project.PeripheralPropertyType;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

public class PresetProperty implements Serializable {
    public String scriptId;
    public PeripheralPropertyType type;
    public Object value;

    public boolean LoadFromJSON(JSONObject jsonObject)
    {
        try {
            this.scriptId = jsonObject.has("scriptId") ? jsonObject.getString("scriptId") : null;
            this.type = jsonObject.has("type") ? PeripheralPropertyType.FromValue(jsonObject.getInt("type")) : PeripheralPropertyType.None;
            this.value = jsonObject.has("value") ? PeripheralPropertyType.Parse(jsonObject.getString("value"), this.type) : null;

        } catch (JSONException e) {
            e.printStackTrace();
            return false;
        }

        return true;
    }

    public JSONObject SaveToJSON()
    {
        JSONObject result = new JSONObject();

        try {
            result.put("scriptId", this.scriptId);
            result.put("type", this.type.GetValue());
            result.put("value", this.value.toString());
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return result;
    }

}

File: Windows/OpenIoT.App/OpenIoT.Lib/Web/Models/Configurations/Presets/PresetProperty.cs
using OpenIoT.Lib.Web.Models.Configurations.Project;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenIoT.Lib.Web.Models.Configurations.Presets
{
    public class PresetProperty
    {
        public string ScriptId { get; set; }
        public PeripheralPropertyType Type { get; set; }
        private object _value;
        public object Value 
        { 
            get { return this._value; }

            set
            {
                switch (this.Type)
                {
                    case PeripheralPropertyType.Integer:
                        this._value = int.Parse(value.ToString());
                        break;
                    case PeripheralPropertyType.Float:
                        this._value = float.Parse(value.ToString());
                        break;
                    case PeripheralPropertyType.Bool:
                        this._value = bool.Parse(value.ToString());
                        break;
                    default:
                        this._value = value;
                        break;
                }
            }
        }
    }
}
